datalake/dlfile.py

```diff
@@ -200,15 +201,14 @@ def to_bundle(self, bundle_filename):
         Args:
         bundle_filename: output file
         '''
-        t = tarfile.open(bundle_filename, 'w')
+        temp_file = tempfile.NamedTemporaryFile(mode='w', delete=False)
+        t = tarfile.open(fileobj=temp_file)
         self._add_fd_to_tar(t, 'content', self._fd)
         self._add_string_to_tar(t, 'version', self.DATALAKE_BUNDLE_VERSION)
         self._add_string_to_tar(t, 'datalake-metadata.json',
                                 self.metadata.json)
         t.close()
-
-        # reset the file pointer in case somebody else wants to read us.
-        self.seek(0, 0)
+        os.rename(temp_file.name, bundle_filename)
```

Comment:
I believe that this is atomic only if the underlying filesystem makes it so, and only if the tmp files happen to be allocated in the same filesystem as the queue directory. A better approach here is to make the uploader ignore, say, files with leading dots. Then we can create the `'.' + bundle_name` files in the queue dir. This will majorly improve the likelihood `os.rename` is atomic.

While you're at it, it might be nice to make the uploader log what appears to be invalid/incomplete bundles and move on.